### DISCUSSION TITLE
folder structure

### DISCUSSION THREAD

**Discussion by ronnyascencio** [masqu3rad3/tik_manager4#125] [Q&A]
Hello, is there a way to know and customize the folder structure of the project? 
if so, where can it be found?

**Comment by masqu3rad3** [comment]
Hi,

Tik Manager4 follows a folder structure which reflects the same inheritence hierarchy pattern of its objects:

Project > SubProject > Task > Category > DCC > Work > Publish

This is the same for the database folders and it is convenient and much safer to maintain the identical structure with database folders.

So unfortunately, it is not possible to change the folder structure without making massive modifications on the core.
On top of that, I would highly recommend not to change it because as I mentioned at the beginning, it reflects the hierarchy pattern. It would be confusing if the hierarchy we see in the Tik Manager UI is different than the folder structure.

By the way, you can create additional folders or file in the root or in other sub-folders. As long as the ones that defined in the database stays there, Tik Manager4 will ignore the extras.

May I ask what is the reason of this customization requirement? Maybe with an example?

**Reply by ronnyascencio** [reply]
Yes I'm trying to make  my write node in nuke  consistent in the output folder : 

For example:

Project/sequence/shot/task/software(nuke)/comp/script/nameofthefile

But when I need to render my final comp or a wip comp my custom write node ( I want ) will search for a folder where to render to maintain structure that's why I ask where can I find the folder structure to see if there is a render folder on tik that I will need to use to set up in the write node in order to render my files there and also want to connect that render file  to sync if is published it will need to go somewhere in a folder 

The same thing for a 3d dcc render for example Houdini where will save the renders in order to make a search tool in nuke to look for the renders of the project coming from cg 

I took a look of the develop doc on tik  but I don't get it quite well lol, so I will be easy to me that inside my Houdini or maya. Can direct with expressions my render to an specific folder ( where tik does not  break ) and in nuke where to render wips/ finals and deliveries 

This is the main thing need to solve 

Because yes tik makes easy to structure a project and creates all scene versioning and that's awesome but need to see where to render from cg and from 2D

**Reply by masqu3rad3** [reply]
As I briefly mentioned on my previous message, it is okay to create your own 'render bin' and put the renders as you want. As long as you don't rename or move around the folders created by Tik Manager, it won't break.

I think you were on the right path for looking the develop extractors docs. However it indeed requires some level of python knowledge and familiarity to api of the relevant DCC.

I would use extractors for all renders, either it is a comp render or 3d renders. This way they would get properly integrated as a publish and protected.

The reason I didn't put anything yet is because it is a bit more flexible area requires adjustments per individual/studio. This also includes heavy interaction of the render farms, which branches out the possibilities massively. That is why I am trying to encourage TDs for custom extractors.

I will put some 'render' extractors soon at least for Nuke, that is in my to-do list. Maybe those can be better starting point templates.
I am also thinking about bringing in the 'sequence viewer' in a more sensible way that I did in Tik Manager v3

**Reply by ronnyascencio** [reply]
Thank you  this helps me a lot to know where to search and how to set up and use the advantages of tik4 as an individual
